hooks/useInfiniteLoading.ts

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useAsyncFn } from 'react-use';
+
+import type { PaginationProps } from 'types';
+
+interface Parameters<T> {
+    fetchingFn: (nextPage: number) => Promise<T[]>;
+    initialData: T[];
+    pagination: PaginationProps;
+}
+
+interface State<T> {
+    canLoadMore: boolean;
+    currentPage: number;
+    data: T[];
+    isLoading: boolean;
+    loadMore: () => void;
+    resetData: () => void;
+}
+
+export function useInfiniteLoading<T>({
```

Comment:
I think we should consider extracting this hook into the theme kit.

Reply:
It depends on presence of an API route in the application, which we can't easily provide as part of theme kit (we can export the handler, but the route still needs to be created). Also, the `include` parameter` here differs from the one in the starter theme and in Bea

Reply:
What do you mean? This hook does not include neither of those things. I think you're talking about `useInfiniteStoriesLoading`?

Reply:
Oh, right! Sorry, I confused the two hooks. This one makes total sense, but since we'll need to update multiple themes, I've logged a separate ticket for that: [TITS-5308](https://linear.app/prezly/issue/TITS-5308/extract-useinfiniteloading-hook-to-theme-kit-nextjs)